src/cdk/table/text-column.ts

```diff
@@ -0,0 +1,136 @@
+ * is using the native table implementation (`<table>`).
+ *
+ * By default, the name of this column will be the header text and data property accessor.
+ * The header text can be overridden with the `headerText` input. Cell values can be overridden with
+ * the `dataAccessor` input. Change the text justification to the start or end using the `justify`
+ * input.
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'cdk-text-column',
+  template: `
+    <ng-container cdkColumnDef>
+      <th cdk-header-cell *cdkHeaderCellDef [style.text-align]="justify">
+        {{headerText}}
+      </th>
+      <td cdk-cell *cdkCellDef="let data" [style.text-align]="justify">
+        {{dataAccessor(data, name)}}
+      </td>
+    </ng-container>
+  `,
+  encapsulation: ViewEncapsulation.None,
+  // Change detection is intentionally not set to OnPush. This component's template will be provided
+  // to the table to be inserted into its view. This is problematic when change detection runs since
+  // the bindings in this template will be evaluated _after_ the table's view is evaluated, which
+  // mean's the template in the table's view will not have the updated value (and in fact will cause
+  // an ExpressionChangedAfterItHasBeenCheckedError).
+  // tslint:disable-next-line:validate-decorators
+  changeDetection: ChangeDetectionStrategy.Default,
+})
+export class CdkTextColumn<T> implements OnDestroy, OnInit {
+  /** Column name that should be used to reference this column. */
+  @Input()
+  get name(): string { return this._name; }
+  set name(name: string) {
+    this._name = name;
+    this.columnDef.name = name;
+  }
+  _name: string;
+
+  /**
+   * Text label that should be used for the column header. If this property is not
+   * set, the header text will default to the column name with its first letter capitalized.
+   */
+  @Input() headerText: string;
+
+  /**
+   * Accessor function to retrieve the data rendered for each cell. If this
+   * property is not set, the data cells will render the value found in the data's property matching
+   * the column's name. For example, if the column is named `id`, then the rendered value will be
+   * value defined by the data's `id` property.
+   */
+  @Input() dataAccessor: (data: T, name: string) => string;
+
+  /** Alignment of the cell values. */
+  @Input() justify: 'start' | 'end' = 'start';
+
+  @ViewChild(CdkColumnDef) columnDef: CdkColumnDef;
+
+  constructor(@Optional() private table: CdkTable<T>,
+              @Optional() @Inject(TEXT_COLUMN_OPTIONS) private options: TextColumnOptions) {
+    this.options = options || {};
+  }
+
+  ngOnInit() {
+    if (this.headerText === undefined) {
+      this.headerText = this._createDefaultHeaderText();
+    }
+
+    if (!this.dataAccessor) {
+      this.dataAccessor = this.options.defaultDataAccessor ||
+          ((data: T, name: string) => (data as any)[name]);
+    }
+
+    if (this.table) {
+      this.table.addColumnDef(this.columnDef);
```

Comment:
Should there be an error if there's no `table` instance?

Reply:
Sure - I can't think of a good reason why it'd be useful otherwise.